### PR TITLE
Fix timm

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*

1. Make timm caps consistent between core and multimodal. Temporarily keep the timm cap in core since the vision module also needs to install timm. We can remove it after refactoring the vision module.
2. Set the default num_classes=0 and compute logits only when `num_classes>0`. This is to handle some special heads in timm, e.g., https://github.com/rwightman/pytorch-image-models/blob/8a1ba67a5019cac4e0e6c430a846e5653f636c3f/timm/models/layers/classifier.py#L38

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by FANGAreNotGnu — APPROVED** [review]
LGTM

**Review by bryanyzhu — APPROVED** [review]
Thanks for the fix, LGTM.

**Review by suzhoum — APPROVED** [review]
LGTM!

**Review by yinweisu — APPROVED** [review]
LGTM!

**Review by sxjscience — APPROVED** [review]
LGTM
